src/03-ticket-calculator.js

```diff
@@ -109,7 +125,52 @@ function calculateTicketPrice(ticketData, ticketInfo) {}
     purchaseTickets(tickets, purchases);
     //> "Ticket type 'discount' cannot be found."
  */
-function purchaseTickets(ticketData, purchases) {}
+function purchaseTickets(ticketData, purchases) {
+  let result = "";// result accumulator 
+  let totalCost = 0;//total cost accumulator 
+  let extrasCost = 0;//extras cost accumulator 
+  let entrantType = "";//place holder for entrant type 
+  let error = "";//place holder for error message to be returned 
+  let extras = "";//extras strings accumulator 
+
+  for (let i = 0; i < purchases.length; i++) {//loop going through an array of purchases 
+    // first we define all possible error messages in case ticket type, entrant type or extras type are not found within ticketData
+    if (ticketData[purchases[i].ticketType] === undefined) { // when that value is  not found it will return as 'undefined' 
+      return "Ticket type 'incorrect-type' cannot be found.";
+    } else if (ticketData[purchases[i].ticketType].priceInCents[purchases[i].entrantType] === undefined) {
+      return "Entrant type 'incorrect-entrant' cannot be found.";
+    } else if (purchases[i].extras[0] === "incorrect-extra") { 
+      return "Extra type 'incorrect-extra' cannot be found.";
+    }
```

Comment:
Same note as above here. Your returns can be more dynamic. Also another approach you could have taken was to leverage the `calculateTicketPrice` function. If the return is a string then your know it is an error message and return that, if not you would add the price for it ticket to a total .